### ISSUE TITLE
Feature Request: Make the entire store readonly

### ISSUE THREAD

**Issue by AlaaEldeenYsr** [mobxjs/mobx.dart#678]
Our project structure depends on immutable state management. All the mobx stores in the project should be immutable, All the observables in the mobx stores should be mutated only from inside the stores, In that case I give readonly annotation for every single observable, I think it would be more much easy if we give only one readonly annotation for the store to make all the observables inside immutable without giving the annotation for every single observable.

**Comment by Ascenio** [comment]
Sounds nice, but there's a catch. What if the class have private dependencies which the class doesn't want to expose? There should be a way to avoid generating getters for them.

**Comment by easeccy** [comment]
I agree. We should discuss it further and it will be a breaking change. 

`@observable` should be equivalent to `Stream<T>`. It should be open for subscription but closed for mutation. Right now it's more likely equivalent of StreamController<T>.
